### ISSUE TITLE
Loaders get delivered to wrong fragment

### ISSUE THREAD

**Issue by padyx** [padyx/purplemoon-android#64]
At some points, pressing the back key or rotating delivers a loader to the wrong fragment, resulting in a ClassCastException:

```
#!java

09-10 20:18:42.857: E/AndroidRuntime(11497): Caused by: java.lang.ClassCastException: ch.defiant.purplesky.util.Holder cannot be cast to ch.defiant.purplesky.beans.UpdateBean
09-10 20:18:42.857: E/AndroidRuntime(11497): 	at ch.defiant.purplesky.activities.MainActivity.onLoadFinished(MainActivity.java:553)
09-10 20:18:42.857: E/AndroidRuntime(11497): 	at android.support.v4.app.LoaderManagerImpl$LoaderInfo.callOnLoadFinished(LoaderManager.java:427)
09-10 20:18:42.857: E/AndroidRuntime(11497): 	at android.support.v4.app.LoaderManagerImpl$LoaderInfo.reportStart(LoaderManager.java:307)
09-10 20:18:42.857: E/AndroidRuntime(11497): 	at android.support.v4.app.LoaderManagerImpl.doReportStart(LoaderManager.java:768)
09-10 20:18:42.857: E/AndroidRuntime(11497): 	at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:581)
09-10 20:18:42.857: E/AndroidRuntime(11497): 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1231)
09-10 20:18:42.857: E/AndroidRuntime(11497): 	at android.app.Activity.performStart(Activity.java:5175)
09-10 20:18:42.857: E/AndroidRuntime(11497): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2433)
09-10 20:18:42.857: E/AndroidRuntime(11497): 	... 12 more

```

**Comment by padyx** [comment]
(Original author of comment is Bitbucket user 'Chakotay')
Loader IDs should be unique - but currently are not

See [Android Reference](https://developer.android.com/reference/android/app/LoaderManager.html#initLoader%28int,%20android.os.Bundle,%20android.app.LoaderManager.LoaderCallbacks%3CD%3E%29)

**Comment by padyx** [comment]
(Original author of comment is Bitbucket user 'Chakotay')
Issue #40 was marked as a duplicate of this issue.

**Comment by padyx** [comment]
(Original author of comment is Bitbucket user 'Chakotay')
first attempt to fix it with commmit  ea64348
see how this holds up

**Comment by padyx** [comment]
(Original author of comment is Bitbucket user 'Chakotay')
See if this is resolved in next version
